docs/source/user_guide/installation/index.rst

```diff
@@ -45,6 +45,10 @@ Dependencies
 ------------
 .. _install.required_dependencies:
 
+.. warning::
+
+    The list of dependencies below might be outdated. Please refer to the ``pyproject.toml`` file to find all dependencies.
```

Comment:
This assumes that users know what `pyproject.toml` (without a link) 😢 

```suggestion
    The list of dependencies below might be outdated. Please use tools like [``pipdeptree``](https://github.com/tox-dev/pipdeptree) to find the latest dependencies of ``pybamm``.
```

@RohitP2005 can you fix this in a new PR if @agriyakhetarpal agrees?

Reply:
Yes, I thought about leaving a permalink to the file, but dropped the idea considering that most users wouldn't require to see it and it might get outdated as well. On second thought, please feel free to add a link to it from the `main` branch, because it's unlikely that the file's location will change, and if it does, it will be noticed caught by improvements like #4686 anyway.

Also, `pipdeptree` does not yet display optional dependencies properly (and neither does `uv pip tree`), so I wouldn't recommend adding it.

Reply:
yeah, i agree with @agriyakhetarpal

Reply:
Nice! @RohitP2005 could you open a PR adding the link?